syntax = "proto3";

package datacloud.clickhouse.v1;

import "datacloud/v1/paging.proto";
import "datacloud/clickhouse/v1/backup.proto";

option go_package = "a.yandex-team.ru/cloud/mdb/datacloud/private_api/datacloud/clickhouse/v1;clickhouse";
option java_package = "doublecloud.priv.clickhouse.v1";
option java_outer_classname = "DCBS";

service BackupService {
    // Returns the specified backup of ClickHouse cluster.
    rpc Get (GetBackupRequest) returns (Backup);

    // Returns the list of available backups for the specified ClickHouse cluster.
    rpc List (ListBackupsRequest) returns (ListBackupsResponse);

    // Delete specified ClickHouse cluster backup.
    rpc Delete (DeleteBackupRequest) returns (DeleteBackupResponse);
}

message GetBackupRequest {
    // Required. ID of the backup to return.
    string backup_id = 1;
}

message ListBackupsRequest {
    // Required. ID of the project to list backups in.
    string project_id = 1;

    // Paging information of the request
    datacloud.v1.Paging paging = 2;
}

message ListBackupsResponse {
    // Requested list of backups.
    repeated Backup backups = 1;

    // Pagination information of the response
    datacloud.v1.NextPage next_page = 2;
}

message DeleteBackupRequest {
    // Required. ID of the backup to delete.
    string backup_id = 1;
}

message DeleteBackupResponse {
    // ID of the operation.
    string operation_id = 1;
}
